examples/example-server-lib/sw_splash.cpp

```diff
@@ -18,92 +18,320 @@
 
 #include "sw_splash.h"
 
-#include <mir/client/display_config.h>
-#include <mir/client/surface.h>
-#include <mir/client/window.h>
-#include <mir/client/window_spec.h>
+#include <wayland-client.h>
 
-#include <mir_toolkit/mir_buffer_stream.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
 
 #include <chrono>
 #include <cstring>
 #include <thread>
 #include <mutex>
+#include <vector>
 
 namespace
 {
-MirPixelFormat find_8888_format(MirConnection* connection)
+template<typename Type>
+auto make_scoped(Type* owned, void(*deleter)(Type*)) -> std::unique_ptr<Type, void(*)(Type*)>
 {
-    unsigned int const num_formats = 32;
-    MirPixelFormat pixel_formats[num_formats];
-    unsigned int valid_formats;
-    mir_connection_get_available_surface_formats(connection, pixel_formats, num_formats, &valid_formats);
+    return {owned, deleter};
+}
+
+struct Globals
```

Comment:
Non blocking, we can always do it later: This struct seems generic enough to be potentially useful to other clients in the project. There might be a better place to put it.